apps/web/src/contexts/communityPage/IsMemberOfCommunityContext.tsx

```diff
@@ -0,0 +1,84 @@
+import { createContext, ReactNode, useCallback, useContext } from 'react';
+import { graphql, useLazyLoadQuery, useRefetchableFragment } from 'react-relay';
+
+import { IsMemberOfCommunityContextFragment$key } from '~/generated/IsMemberOfCommunityContextFragment.graphql';
+import { IsMemberOfCommunityContextQuery } from '~/generated/IsMemberOfCommunityContextQuery.graphql';
+import { IsMemberOfCommunityContextViewerQuery } from '~/generated/IsMemberOfCommunityContextViewerQuery.graphql';
+
+type IsMemberOfCommunityState = {
+  isMemberOfCommunity: boolean;
+  refetchIsMemberOfCommunity: () => void;
+};
+
+export const IsMemberOfCommunityContext = createContext<IsMemberOfCommunityState | undefined>(
+  undefined
+);
+
+export const useIsMemberOfCommunity = (): IsMemberOfCommunityState => {
+  const context = useContext(IsMemberOfCommunityContext);
+  if (!context) {
+    throw new Error('Attempted to use IsMemberOfCommunityContext without a provider');
+  }
+
+  return context;
+};
+
+type IsMemberOfCommunityProviderProps = {
+  children: ReactNode;
+  communityDbid: string;
+};
+
+export const IsMemberOfCommunityProvider = ({
```

Comment:
I decided to use a context because `isMemberOfCommunity` is used in different areas of the Community Page that are a bit separate from one another, and I wanted to avoid them having to make redundant separate requests to retrieve this value. isMemberOfCommunity is its own query because it requires a community id which we don't have until our first query for community data is  resolved.

Reply:
That make sense! No need to do prop drilling too

Reply:
sweet, I just added my comment as a comment to the file lol